R/backport_linter.R

```diff
@@ -0,0 +1,83 @@
+#' @describeIn linters that checks for usage of unavailable functions. Not reliable for testing r-devel dependencies.
+#' @export
+backport_linter <- function(r_version = "4.0.3") {
+  function(source_file) {
+    if (is.null(source_file$xml_parsed_content) || r_version >= "r-devel") return(list())
```

Comment:
Why is there a lexicographic comparison to "r-devel"?
Both "r-release" and "r-oldrel" would trigger this.

Reply:
Maybe I should just leave this as `"r"` -- I don't think we want to support tracking devel dependencies, I just wanted something that would be bigger than all R version numbers (at first I used `"999999"` as the name, `"r-devel"` felt better but you raise a good point)

Reply:
Couldn't we consult `names(backports)` to find whether the selected version is supported?
The highest of those versions wouldn't cause lints anyway so we can `tail()` it to obtain a definitive list of R versions that can result in lints if I'm not mistaken.

Reply:
if so we'd have to include in the list all released versions (e.g. 3.5.3).

And sorry I'm just waking up, I'm not following your second point

Reply:
Good point.
Wouldn't we never lint if `r_version` is at least `getRversion()`?
i.e. `if (r_version >= getRversion()) return(list())`.
That way if r_version is older than the current environment, backport problems can be detected.
If it isn't there are no backports that can be used from the current R version which would be a problem in `r_version`, right?

Reply:
I think that's _basically_ right, but I do think it's possible to run this on a machine that has a different environment than required. Say I work on two machines and the one I'm using today has an old version. We wouldn't want to skip then necessarily right?

Reply:
On the old version, the used backports would either break the code completely (which you'd notice), or the backports were already safely ported to your old version. No?

Reply:
That's if we run tests, but `lintr` is static analysis so we could in principle run it on a different machine that doesn't execute code.

If that machine is running for a bigger library of R code (say a local CRAN), there may be different code requirements.

I think it's basically an edge case... but taking a step back, the goal of this is to get around/change the `r_version >= "r"` part? I'm happier with that now than I was with `r_version >= "r-devel"`

Reply:
I agree. Let's just leave it as-is for now.
Maybe later become smart about what `r_version`s are found in `backports` and thus will potentially cause lints.